blocks/bricks/cost.py

```diff
@@ -63,10 +61,49 @@ def apply(self, y, y_hat):
 
 
 class MisclassificationRate(Cost):
+    """Calculates the misclassification rate for a mini-batch.
+
+    Parameters
+    ----------
+    top_k : int, optional
+        If the ground truth class is within the `top_k` highest
+        responses for a given example, the model is considered
+        to have predicted correctly. Default: 1.
+
+    Notes
+    -----
+    Ties for `top_k`-th place are broken pessimistically, i.e.
+    in the (in practice, rare) case that there is a tie for `top_k`-th
+    highest output for a given example, it is considered an incorrect
+    prediction.
+
+    """
+    def __init__(self, top_k=1):
+        self.top_k = top_k
+        super(MisclassificationRate, self).__init__()
+
     @application(outputs=["error_rate"])
     def apply(self, y, y_hat):
+        # Support checkpoints that predate self.top_k
+        top_k = getattr(self, 'top_k', 1)
+        if top_k == 1:
+            mistakes = tensor.neq(y, y_hat.argmax(axis=1))
+        else:
+            flattened_responses = y_hat.reshape((y_hat.shape[0] *
```

Comment:
Heh, actually wasn't sure that existed in Theano. Pushed a new version with that change and the `.mean()` change.